arch/x86/adler32_avx512.c

```diff
@@ -15,13 +15,37 @@
 
 #ifdef X86_AVX512_ADLER32
 
+#ifdef X86_AVX2_ADLER32 
+extern uint32_t adler32_avx2(uint32_t adler, const unsigned char *buf, size_t len);
+#endif
+#ifdef X86_SSE41_ADLER32 
+extern uint32_t adler32_sse41(uint32_t adler, const unsigned char *buf, size_t len);
+#endif
+
 Z_INTERNAL uint32_t adler32_avx512(uint32_t adler, const unsigned char *buf, size_t len) {
     uint32_t sum2;
 
+    /* For impossibly tiny sizes, use the smaller width versions. We still need
+     * to check for compile time support for these but they are likely there */
+#ifdef X86_SSE41_ADLER32 
+    if (len < 32) {
```

Comment:
I'm not sure if these braces are needed.

Reply:
They aren't

Reply:
I recommend for these changes removing the braces if it is just a one-liner. I say that because other areas and previous code even in adler don't do it.